### DISCUSSION TITLE
Storage and retrieval markets

### DISCUSSION THREAD

**Discussion by zettai-caiman** [ChainSafe/forest#2304] [Ideas]
Discussion about implementation of markets into Forest 🌲

**Comment by zettai-caiman** [comment]
My general thoughts for beta release is that we focus on retrieval markets. From my previous conversation with Lotus team, they are focusing on storage already. @lemmih

**Reply by q9f** [reply]
We should certainly have both, but not sure we should have this in beta already.

**Reply by zettai-caiman** [reply]
Ya, it's really very costly in terms of infrastructure to run a storage miner. A lot of other projects (including Lotus & Venus) is already focusing on storage providers. In order to add value to the FIL ecosystem and from a differentiation angle we were aiming for retrieval markets, FVM developers and a light client.
